components/constellation/constellation.rs

```diff
@@ -3239,13 +3267,22 @@ where
 
         match old_pipeline_id {
             None => {
+                let new_context_info = match change.new_browsing_context_info {
+                    Some(new_context_info) => new_context_info,
+                    None => {
+                        return warn!(
```

Comment:
It's unfortunate that we have to add this check here, but I'm not sure of a way around it at the moment.

Reply:
Yeah, I agree. While overall the introduction of `NewBrowsingContextInfo` makes the code cleaner and more explicit, this is an unfortunate side effect.
It could be solved by moving `parent_pipeline_id` back to `SessionHistoryChange` and always setting it so that it's available in this function, but that kind of muddies the semantics, since most of the time the field isn't actually needed by whoever uses `SessionHistoryChange`.